### PR TITLE
frontend

### PR DESCRIPTION
**Pull request by oliverye7**
1. set up basic frontend routes (/login, /play, /home) without any css styling
2. connect frontend to backend
3. update backend to allow CORS from localhost, remove /proto directory

### CODE REVIEW

File: backend/src/api/challenge.rs [hunk 1]
```diff
@@ -21,8 +21,11 @@ pub struct ChallengeResponse {
 pub async fn get_challenge() -> Json<ChallengeResponse> {
     info!("Requesting to get a challenge.");
     let response = ChallengeResponse {
-        message: "User requested to get a challenge".to_string(),
+        message: "received challenge: lorem ipsum dolor... ".to_string(),
     };
+    //let response = ChallengeResponse {
+    //    message: "User requested to get a challenge".to_string(),
+    //};
```
Comment (rahulk29) [inline backend/src/api/challenge.rs:28]: delete?

**Review by rahulk29 — APPROVED** [review]

File: client/src/index.html
Comment (seany4ng) [inline client/src/index.html:1]: delete this file

**Review by seany4ng — CHANGES_REQUESTED** [review]
only blocking changes are to revive the `.gitignore` in backend so the next PR doesn't add a bunch of random shit, and remove `/src/index.html` before we forget that it's useless. rest lgtm

File: client/nginx/nginx.conf
[the remark this answers is not in the document]
Comment (oliverye7) [reply client/nginx/nginx.conf:1]: its literally "nginx" tho
https://nginx.org/en/

[the remark this answers is not in the document]
Comment (seany4ng) [reply client/nginx/nginx.conf:1]: but what is it being used for rn, i don't see any references to this in the actual code

File: client/src/components/Home/Body.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <a href="/login">
+          <button>login</button>
+        </a>
+      </div>
+      <div>
+        <a href="/play">
+          <button className="">play</button>
+        </a>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (oliverye7) [reply client/src/components/Home/Body.js:17]: yeah this is going to all go in the `tailwind.config` file which should be created already, haven't decided on site colors yet so for now we'll keep that mostly empty 

right now `tailwind.config` just contains the font karla that i blanket applied cuz i used it for my other site

**Review by seany4ng — APPROVED** [review]

**Pull request merged**